File: Buddha-Hr-frontend/buddha-hr-frontend/src/services/userService.js
import api from "./api";

export const getUsers = async (
  page = 0,
  size = 10,
  sortBy = "createdAt",
  direction = "desc"
) => {

  const response = await api.get("/users", {
    params: {
      page,
      size,
      sortBy,
      direction,
    },
  });

  return response.data;
};

export const getUser = async (id) => {

  const response = await api.get(`/users/${id}`);

  return response.data;
};

export const updateUser = async (
  id,
  data
) => {

  const response = await api.put(
    `/users/${id}`,
    data
  );

  return response.data;
};

export const deleteUser = async (id) => {

  await api.delete(`/users/${id}`);
};

export const updateUserRole = async (id, role) => {

  const response = await api.patch(
    `/users/${id}/role`,
    {
      role,
    }
  );

  return response.data;

};

File: buddha-hr-backend/buddha-hr-backend/src/main/java/com/buddhahr/user/service/UserService.java
package com.buddhahr.user.service;

import com.buddhahr.user.dto.request.RegisterRequest;
import com.buddhahr.user.dto.request.UpdateUserRequest;
import com.buddhahr.user.dto.request.UpdateUserRoleRequest;
import com.buddhahr.user.dto.response.UserResponse;
import org.springframework.data.domain.Page;

public interface UserService {

    UserResponse register(RegisterRequest request);

    Page<UserResponse> getAllUsers(
            int page,
            int size,
            String sortBy,
            String direction
    );

    UserResponse getUserById(Long id);

    UserResponse updateUser(
            Long id,
            UpdateUserRequest request
    );
    UserResponse updateUserRole(
            Long id,
            UpdateUserRoleRequest request
    );

    void deleteUser(Long id);
}

